### PR TITLE
[chore](compile) fix compile error in ColumnObject

### PR DESCRIPTION
**Pull request by eldenmoon**
## Proposed changes

This is issue is caused by the two PR merged without conflict
#27557
#27676

Issue Number: close #xxx

## Further comments

If this is a relatively large or complex change, kick off the discussion at [dev@doris.apache.org](mailto:dev@doris.apache.org) by explaining why you chose the solution you did and what alternatives you considered, etc...

### CODE REVIEW

**Comment by eldenmoon** [comment]
run buildall

File: be/src/vec/columns/column_object.cpp [hunk 1]
```diff
@@ -1435,13 +1435,13 @@ void ColumnObject::insert_indices_from(const IColumn& src, const int* indices_be
...
                                             const uint32_t* indices_end) {
-    // insert_indices_from with alignment
-    const ColumnObject& src_column = *check_and_get_column<ColumnObject>(src);
-    align_variant_by_name_and_type(*this, src_column, indices_end - indices_begin,
-                                   [indices_begin, indices_end](const IColumn& src, IColumn* dst) {
-                                       dst->insert_indices_from_join(src, indices_begin,
-                                                                     indices_end);
-                                   });
+    for (auto x = indices_begin; x != indices_end; ++x) {
+        if (*x == 0) {
```
Comment (BiteTheDDDDt) [inline be/src/vec/columns/column_object.cpp:1439]: remove x==0 branch

Comment (eldenmoon) [reply be/src/vec/columns/column_object.cpp]: done

**Comment by eldenmoon** [comment]
run buildall

**Review by BiteTheDDDDt — APPROVED** [review]

**Review by qidaye — APPROVED** [review]
LGTM

**Pull request merged**